test/e2e/thanosruler_test.go

```diff
@@ -258,3 +259,90 @@ func testTRMinReadySeconds(t *testing.T) {
 		t.Fatalf("expected MinReadySeconds to be %d but got %d", updated, trSS.Spec.MinReadySeconds)
 	}
 }
+
+// Tests Thanos ruler -> Alertmanger path
+// This is done by creating a firing rule that will be picked up by
+// Thanos Ruler which will send it to Alertmanager, finally we will
+// use the Alertmanager API to validate that the alert is there
+func testTRAlertmanagerConfig(t *testing.T) {
+	const (
+		name       = "test"
+		group      = "thanos-alertmanager-test"
+		secretName = "thanos-ruler-alertmanagers-config"
+		configKey  = "alertmanagers.yaml"
+		testAlert  = "alert1"
+	)
+	testCtx := framework.NewTestCtx(t)
+	defer testCtx.Cleanup(t)
+
+	ns := framework.CreateNamespace(context.Background(), t, testCtx)
+	framework.SetupPrometheusRBAC(context.Background(), t, testCtx, ns)
+
+	// Create Alertmanager resource and service
+	alertmanager, err := framework.CreateAlertmanagerAndWaitUntilReady(context.Background(), ns, framework.MakeBasicAlertmanager(name, 3))
+	assert.NoError(t, err)
+
+	amSVC := framework.MakeAlertmanagerService(alertmanager.Name, group, v1.ServiceTypeClusterIP)
+	_, err = framework.CreateOrUpdateServiceAndWaitUntilReady(context.Background(), ns, amSVC)
+	assert.NoError(t, err)
+
+	// Create a Prometheus resource because Thanos ruler needs a query API.
+	prometheus, err := framework.CreatePrometheusAndWaitUntilReady(context.Background(), ns, framework.MakeBasicPrometheus(ns, name, name, 1))
+	assert.NoError(t, err)
+
+	svc := framework.MakePrometheusService(prometheus.Name, name, v1.ServiceTypeClusterIP)
+	_, err = framework.CreateOrUpdateServiceAndWaitUntilReady(context.Background(), ns, svc)
+	assert.NoError(t, err)
+
+	// Create Secret with Alermanager config,
+	trAmConfigSecret := &v1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: secretName,
+		},
+		Data: map[string][]byte{
+			configKey: []byte(fmt.Sprintf(`
+alertmanagers:
+- scheme: http
+  api_version: v2
+  static_configs:
+    - dnssrv+_web._tcp.%s.%s.svc.cluster.local
+`, amSVC.Name, ns)),
+		},
+	}
+	_, err = framework.KubeClient.CoreV1().Secrets(ns).Create(context.Background(), trAmConfigSecret, metav1.CreateOptions{})
+	assert.NoError(t, err)
+
+	// Create Thanos ruler resource and service
+	thanos := framework.MakeBasicThanosRuler(name, 1, fmt.Sprintf("http://%s:%d/", svc.Name, svc.Spec.Ports[0].Port))
+	thanos.Spec.RuleSelector = &metav1.LabelSelector{}
```

Comment:
this isn't directly related to your PR but `MakeBasicThanosRuler()` could already set the rule selector to empty.